### DISCUSSION TITLE
Variable in "Value Mappings", "Thresholds"

### DISCUSSION THREAD

**Discussion by devpakk** [grafana/grafana#58834] [Dashboards, Panels & Visualizations]
Good afternoon!

Please consider improving Grafana and adding the possibility of using variables in "Value Mappings". At the moment, adding to Threshhold is not possible, and when used in "Value Mappings", the value is not applied.

Adding to variable to "Value Mappings", step by step:

1. Create a dashboard
2. Create a variable
3. Use this variable in Value Mappings to replace the corresponding regular expression

It is very useful when implementing dynamic dashboards.

Ex:
![image](https://user-images.githubusercontent.com/99316743/202202118-8205fd38-29c3-451a-9b17-f6626dc1a9b6.png)

**Comment by brettdaman** [comment]
+1

**Comment by kallaics** [comment]
Hi guys,
I mean it would be a really great feature in Grafana. I am facing with same issue and I hope this feature will be implemented.
I found the original discussion with @usmangt about the topic: https://community.grafana.com/t/variable-with-value-mappings/24543/10

Thanks for all of your efforts.

**Reply by HorselessName** [reply]
Hello. Hey, do you know how to use the RegEx in the value mappings you provided in the screenshot to replace a specific part of the text only? I see that you used .* to match the entire thing and replace it to $ds2. What if you just want to replace a string? Lets say you have "This is a Service" and want to replace the word "Service" with "Software"...

I my case, I have "Status do Serviço "TOTVSservice"" and want to remove "Status do Serviço."

Here is how I did it but it did not work:
(^Status do Serviço\s*)

Image of it:

![image](https://user-images.githubusercontent.com/20173635/221371030-bdd0b147-c83a-4747-9907-f57bf00ba17c.png)

**Comment by megaak** [comment]
+1. This would be incredibly useful.

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was converted into an issue, which is available [here](https://github.com/grafana/grafana/issues/73439).

**Comment by YerBabiE** [comment]
+1, it would be super useful to show the actual value and then alongside some text to see if it's normal, low etc. At the moment there is no easy way to see what the value is at a glance. i.e. low, but how low?
